Mathlib/LinearAlgebra/CliffordAlgebra/Conjugation.lean

```diff
@@ -190,89 +190,72 @@ variable (Q)
 section Involute
 
 theorem submodule_map_involute_eq_comap (p : Submodule R (CliffordAlgebra Q)) :
-    p.map (involute : CliffordAlgebra Q →ₐ[R] CliffordAlgebra Q).toLinearMap =
-      p.comap (involute : CliffordAlgebra Q →ₐ[R] CliffordAlgebra Q).toLinearMap :=
+    p.map involute = p.comap involute :=
```

Comment:
mathlib3#16105 was the PR that made `Submodule.(co)map` accept `SemilinearMapClass`, and the author @dupuisf explained his motivation [there](https://github.com/leanprover-community/mathlib3/pull/16105#issuecomment-1219588209). Overall indeed I don't see much advantage of this generality, and it seems to me that using the coercion to `SemilinearMap` would be good enough. So my plan now is to leave this PR to you and make my other PR independent of this one. Let me also bring the reviewer of the original PR @Vierkantor to the discussion.

The change (in Lean 3) required specifying the type of `involute` here, but now it's no longer necessary (if you write `involte.toLinearMap` it still works, so this is something we can golf regardless of the decision whether to remove the `.toLinearMap`. One could be tempted to replace `.toLinearMap` by coercions, but then you probably have to specify the type. IIRC the distinction between `.toLinearMap` and coercion can require `erw`s.